framework/src/main/java/org/checkerframework/common/initializedfields/InitializedFieldsAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,235 @@
+         *
+         * @param factory the type factory associated with the newly-created ContractsUtils
+         */
+        public InitializedFieldsContractsUtils(GenericAnnotatedTypeFactory<?, ?, ?, ?> factory) {
+            super(factory);
+        }
+
+        @Override
+        public Set<Contract.Postcondition> getPostconditions(ExecutableElement executableElement) {
+            Set<Contract.Postcondition> result = super.getPostconditions(executableElement);
+
+            // Only process methods defined in source code being type-checked.
+            if (declarationFromElement(executableElement) != null) {
+
+                if (executableElement.getKind() == ElementKind.CONSTRUCTOR) {
+                    // It's a constructor
+
+                    String[] fieldsToInitialize =
+                            fieldsToInitialize(
+                                    (TypeElement) executableElement.getEnclosingElement());
+                    if (fieldsToInitialize.length != 0) {
+
+                        AnnotationMirror initializedFieldsAnno;
+                        {
+                            AnnotationBuilder builder =
+                                    new AnnotationBuilder(
+                                            processingEnv,
+                                            InitializedFields.class.getCanonicalName());
+                            builder.setValue("value", fieldsToInitialize);
+                            initializedFieldsAnno = builder.build();
+                        }
+                        AnnotationMirror ensuresAnno;
+                        {
+                            AnnotationBuilder builder =
+                                    new AnnotationBuilder(
+                                            processingEnv,
+                                            EnsuresInitializedFields.class.getCanonicalName());
+                            builder.setValue("value", new String[] {"this"});
+                            builder.setValue("fields", fieldsToInitialize);
+                            ensuresAnno = builder.build();
+                        }
+                        Contract.Postcondition ensuresContract =
+                                new Contract.Postcondition(
+                                        "this", initializedFieldsAnno, ensuresAnno);
+
+                        result.add(ensuresContract);
+                    }
+                }
+            }
+
+            return result;
+        }
+    }
+
+    /**
+     * Returns the fields that the constructor must initialize. These are the fields F declared in
+     * this class that satisfy all of the following conditions:
+     *
+     * <ul>
+     *   <li>F is a non-final field (if final, Java will issue a warning, so we don't need to).
+     *   <li>F's declaration has no initializer.
+     *   <li>No initialization block or static initialization block sets the field. (This seems to
```

Comment:
I think these fields are still included in the returned array.  It's just that dataflow visits initialization block or static initialization block as part of the constructor.  So, I would delete this bullet point.